### ISSUE TITLE
Default to always plot on raw scale for both outcomes and predictors?

### ISSUE THREAD

**Issue by mattansb** [strengejacke/ggeffects#154]
tl;dr:
- default to plot raw predictors / outcomes?
- label transformed (not raw) predictors
- transform data

---

``` r
library(ggeffects)


fit <- lm(mpg ~ log(hp), mtcars)
```

Was wondering why is this not the default beavior?

``` r
plot(ggeffect(fit, "hp [exp]"))
#> Loading required namespace: ggplot2
```

![](https://i.imgur.com/ydeYSqH.png)

The default behavior for `predict()` and `emmeans` is to use the “raw” scales
of predictors and outcomes.

``` r
emmeans::emmip(fit, ~hp, CIs = T,
               at = list(hp = seq(50,300, by = 50)))
```

![](https://i.imgur.com/bgC3nqI.png)

``` r

plot(seq(50,300, by = 50),
     predict(fit, newdata = data.frame(hp = seq(50,300, by = 50))))
```

![](https://i.imgur.com/HiHNGw6.png)

Also, I think the label for transformed vars should indicate they are transformed, so here this should be `log(hp)` on the x-axis.

``` r
plot(ggeffect(fit, "hp"))
```

![](https://i.imgur.com/NTCBG4t.png)

(This does happen when the outcome is transformed, but here too I think the default behavior should be to show the raw scale, as is done with GLMs etc…)

Also note the raw data does not get transformed:

``` r
plot(ggeffect(fit, "hp [exp]"), add.data = TRUE)
```

![](https://i.imgur.com/0EhGmtF.png)

<sup>Created on 2020-08-21 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by strengejacke** [comment]
good point! I wonder whether `log()` is the only transformation that should be caught?

**Comment by mattansb** [comment]
At least the common ones?
log, log1p, exp, expm1, scale, sqrt...

**Comment by strengejacke** [comment]
> The default behavior for predict() and emmeans is to use the “raw” scales of predictors and outcomes.

In your example, you specify the `at` values. In this meaning, `ggpredict()` and `predict()` / `emmeans()` behave identical.

For each function, you need to specify the value range:
``` r
library(ggeffects)
library(emmeans)

fit <- lm(mpg ~ log(hp), mtcars)

plot(ggpredict(fit, "hp [50:300 by=50]"))
#> Model has log-transformed predictors. Consider using `terms="hp [exp]"` to back-transform scale.
#> Loading required namespace: ggplot2
```

![](https://i.imgur.com/kOrTdE8.png)

``` r
plot(seq(50,300, by = 50), predict(fit, newdata = data.frame(hp = seq(50,300, by = 50))))
```

![](https://i.imgur.com/Nc3e1FP.png)

``` r
plot(emmip(fit, ~hp, at = list(hp = seq(50,300, by = 50))))
```

![](https://i.imgur.com/bzrc4z9.png)

<sup>Created on 2020-08-26 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

So I'm not sure if there's really the need to change the behaviour here? Though the raw data points indeed should be adjusted.

**Comment by mattansb** [comment]
Ah, I think something else is happening here!

All predictions are on the raw scale (via `emmeans` / `predict` / `effects`), but the "shortcut" options are taken from the transformed scale!

``` r
library(ggeffects)

fit <- lm(mpg ~ log(hp), mtcars)

plot(ggpredict(fit, "hp"))
#> Model has log-transformed predictors. Consider using `terms="hp [exp]"` to back-transform scale.
#> Loading required namespace: ggplot2
```

![](https://i.imgur.com/3Jgb3t7.png)

``` r
plot(ggpredict(fit, "hp [3.95:5.81 by=0.01]"))
#> Model has log-transformed predictors. Consider using `terms="hp [exp]"` to back-transform scale.
```

![](https://i.imgur.com/48ZumzS.png)

<sup>Created on 2020-08-26 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>



So the x-axis is **always** the raw scale, but the "shortcut" options are incorectly looking at the transformed scale!
This means that what `plot(ggpredict(fit, "hp [exp]"))` does is back-transform the predictor before using the "shortcut"s, but the prediction is, regardless of this, always treating it as on the raw.

It seems that this is because [`insight::get_predictors` returns the transformed data and not the raw data](https://github.com/easystats/insight/issues/216) 😱

I hope I am describing this bug clearly :/

**Comment by strengejacke** [comment]
Now since https://github.com/easystats/insight/issues/216 is resolved, this issue seems to be resolved as well... I just need to remove the warning message:

``` r
library(ggeffects)
fit <- lm(mpg ~ log(hp), mtcars)
plot(ggpredict(fit, "hp"), add.data = TRUE)
#> Model has log-transformed predictors. Consider using `terms="hp [exp]"` to back-transform scale.
```

![](https://i.imgur.com/EwrXICD.png)

<sup>Created on 2020-08-27 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by strengejacke** [comment]
``` r
library(ggeffects)
fit <- lm(log(log(mpg)) ~ log(hp), mtcars)
plot(ggpredict(fit, "hp"), add.data = TRUE)
#> Model has log-transformed response. Back-transforming predictions to original response scale. Standard errors are still on the log-scale.
```

![](https://i.imgur.com/i3iJZa9.png)

<sup>Created on 2020-08-27 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

I think we can close this, no?

**Comment by mattansb** [comment]
This looks awesome!

Thanks for the hard work! 🎈
